aten/src/ATen/native/mps/OperationUtils.mm

```diff
@@ -1002,7 +1003,7 @@ static dispatch_data_t getSectionData(const std::string& name) {
       getMPSProfiler().endProfileKernel(cplState);
     });
   }
-  if (needs_output_copy) {
+  if (!same_strides) {
```

Comment:
This is also wrong, most commonly input is sliced, but output is dense, so no need for an extra copy here